HeterogeneousCore/SonicTriton/interface/TritonData.h

```diff
@@ -47,7 +47,9 @@ class TritonData {
     converterName_ = conf.getParameter<std::string>("converterName");
   }
   template <typename DT>
-  std::unique_ptr<TritonConverterBase<DT>> createConverter() const { return TritonConverterFactory<DT>::get()->create(converterName_); }
+  void createConverter() const { 
+    if (!converter_.has_value()) converter_ = std::shared_ptr<TritonConverterBase<DT>>(TritonConverterFactory<DT>::get()->create(converterName_));
```

Comment:
This checks if it's filled, but not if it's the right type. (I don't anticipate a given TritonData object being used for more than one primitive type, but we should try to be future-proof.) I propose the following (also to reduce type name repetition):
```cpp
  using ConverterType = std::shared_ptr<TritonConverterBase<DT>>;
  if (std::any_cast<ConverterType>(converter_)==nullptr) converter_ = ConverterType(TritonConverterFactory<DT>::get()->create(converterName_));
```

and actually, now I've realized that the member variable has to be `std::shared_ptr<std::any>`, because `any` has to contain a bare pointer in order for `any_cast` to return `nullptr` when the type is wrong. So it needs a little more revision than my above suggestion; let me know if this isn't clear.

Reply:
In order to catch both the empty and wrong-type conversions, I think we want to keep `std::any` to store things. I tried with `std::shared_ptr<std::any>` and had some issues, and this seems to be what we need anyway